akka-docs/src/test/java/jdocs/stream/IntegrationDocTest.java

```diff
@@ -770,4 +770,26 @@ public void illustrateSourceQueue() throws Exception {
       }
     };
   }
+
+  @Test
+  public void illustrateSourceActorRef() throws Exception {
+    new TestKit(system) {
+      {
+        // #source-actorRef
+        int bufferSize = 10;
+
+        Source<Integer, ActorRef> source =
+            Source.actorRef(
+                bufferSize, OverflowStrategy.dropHead()); // note: backpressure is not supported
+        ActorRef actorRef =
+            source.map(x -> x * x).to(Sink.foreach(x -> System.out.println("got: " + x))).run(mat);
+
+        actorRef.tell(1, ActorRef.noSender());
```

Comment:
Perhaps use something like 'self' here? Should be fairly rare to use the `ActorRef` integration when not actually in an actor context ;)

Reply:
I never used Akka from Java, just relied on the other java examples for this.